packages/components/src/components/post-collapsible/post-collapsible.tsx

```diff
@@ -0,0 +1,115 @@
+import { Component, Element, h, Method, Prop, State } from '@stencil/core';
+import { checkValidHTMLTag, getElementHeight, onTransitionEnd } from '../../utils';
+
+let nextId = 0;
+
+@Component({
+  tag: 'post-collapsible',
+  styleUrl: 'post-collapsible.scss',
+  shadow: true,
+})
+export class PostCollapsible {
+  /**
+   * If `true`, the element is initially collapsed otherwise it is displayed.
+   */
+  @Prop({ mutable: true }) collapsed?: boolean = false;
+
+  /**
+   * Defines the hierarchical level of the collapsible header within the headings structure.
+   */
+  @Prop() headingLevel?: number = 2;
+
+  @State() collapseClasses: string;
+  @State() collapseHeight: string | null = null;
+
+  @Element() host: HTMLElement;
+
+  hasHeader: boolean;
+  headingTag: string | undefined;
+  collapsibleId: string;
+  collapsibleElement: HTMLElement;
+
+  componentWillLoad() {
+    this.hasHeader = !!this.host.querySelector('[slot="header"]');
```

Comment:
Users could use slot-binding on multiple elements.

```suggestion
    this.hasHeader = this.host.querySelectorAll('[slot="header"]').length > 0;
```